erpnext/manufacturing/doctype/production_plan/production_plan.js

```diff
@@ -121,14 +121,20 @@ frappe.ui.form.on('Production Plan', {
 	},
 
 	make_work_order: function(frm) {
-		frappe.call({
-			method: "make_work_order",
-			freeze: true,
-			doc: frm.doc,
-			callback: function() {
-				frm.reload_doc();
-			}
-		});
+		if(frm.doc.status === 'Material Received') {
+			frappe.call({
+				method: "make_work_order",
+				freeze: true,
+				doc: frm.doc,
+				callback: function() {
+					frm.reload_doc();
+				}
+			});
+		}
+		else {
+            frappe.msgprint("Requested Material not yet received")
+        }
+
```

Comment:
Remove this change
```suggestion

```
and 
Add at line number 61
```suggestion
		if (frm.doc.docstatus === 1 && frm.doc.po_items
			&& !in_list(['Material Received', 'Completed'], frm.doc.status)) {
			frm.add_custom_button(__("Work Order"), ()=> {
				frm.trigger("make_work_order");
			}, __('Create'));
		}

```